### DISCUSSION TITLE
(Solved)AIO working with Cloudflare?

### DISCUSSION THREAD

**Discussion by H3is3nb3rg404** [nextcloud/all-in-one#2845] [Questions] [answered]
Hey all,
I have Nextcloud working or at least the containers are up and running. However, I cannot access Nextcloud either via local Ip:11000
Or through Cloudflare, I only get an error code 502 bad Gateway.

I'm running on Synology NAS.
Anyone able to get this to work?
I also have TLS enabled in Cloudflare for my tunnel.

Thank you to all that can help!

**Comment by szaimen** [comment]
Hi, see https://github.com/nextcloud/all-in-one/blob/main/reverse-proxy.md#6-how-to-debug-things

**Comment by vawaver** [comment]
@H3is3nb3rg404 
Did you manage to figure out how to set it up correctly?
I am currently struggling with the same problem.
Cloudflare tunnel setup + correct setup for AIO docker-compose.yml

**Reply by H3is3nb3rg404** [reply]
No, I could never get Cloudflare to work.....yet

**Comment by vawaver** [comment]
@H3is3nb3rg404 
I found out how to set it up. It took a couple of setting testing but I handled it yesterday.
Now I know it does not take more than 15 minutes.

**Comment by vawaver** [comment]
> No, I could never get Cloudflare to work.....yet

I use Docker version of the Cloudflare connector.
Example of Cloudflare Docker connector settings : 

`docker run cloudflare/cloudflared:latest tunnel --no-autoupdate run --token eyJhIjoiZDdiYWVkY2M4YTc2MGIzMWI2ZTFlYmMxNzNlMTliNWIiLCJ0IjoiNGFkOWE2ZWYtOTc5Yy00ZDg1LWI2MGQtMzU0NzRmNDJkYTEzIiwicyI6Ik1HSTBOR1EyWTJZdFlUZGlNUzAwT0dRM0xUazBOMlV0TmpVMlpqQmxNVFUxTlRsaiJ9`

You had to add these parameters to the command
`-d --name cloudflaretunnel --restart unless-stopped`

The command is now:`
docker run -d --name cloudflaretunnel --restart unless-stopped cloudflare/cloudflared:latest tunnel --no-autoupdate run --token eyJhIjoiZDdiYWVkY2M4YTc2MGIzMWI2ZTFlYmMxNzNlMTliNWIiLCJ0IjoiNGFkOWE2ZWYtOTc5Yy00ZDg1LWI2MGQtMzU0NzRmNDJkYTEzIiwicyI6Ik1HSTBOR1EyWTJZdFlUZGlNUzAwT0dRM0xUazBOMlV0TmpVMlpqQmxNVFUxTlRsaiJ9`
``

Cloudlfare docker connector starts in the background.

**Reply by H3is3nb3rg404** [reply]
Sorry, Internet was down all day.
I already have a Cloudflare container running do I need to add this to the existing container?
or
Do I need to remove the old configuration and install the new command?

**Reply by aaronsbytes** [reply]
This did it for me. Thank you so much!

**Comment by vawaver** [comment, marked as answer]
@H3is3nb3rg404 
If you have a working Cloudflare injector let me share my settings for other.
Here is a short tutorial.

**SOURCES**
OS: Ubuntu server 22.04.02
Local IP address of the device: 192.168.1.31
Domain: cloud.battlezilla.com

**FIREWALL**
It may be necessary to enable ports on the device 8080 and 11000
```
sudo ufw allow 8080
sudo ufw allow 11000
sudo ufw reload
```
**NEXTCLOUD AIO**
Create a directory and in it the docker-compose.yml file
`mkdir nextcloud-aio && cd nextcloud-aio && nano docker-compose.yml`

insert code
```

version: "3.8"

services:
  nextcloud:
    image: nextcloud/all-in-one:latest
    restart: always
    container_name: nextcloud-aio-mastercontainer 
    volumes:
      - nextcloud_aio_mastercontainer:/mnt/docker-aio-config 
      - /var/run/docker.sock:/var/run/docker.sock:ro 
    ports:
      - 80:80 
      - 8080:8080
    environment: 
      - APACHE_PORT=11000 
      - APACHE_IP_BINDING=0.0.0.0
      - SKIP_DOMAIN_VALIDATION=true

volumes:
  nextcloud_aio_mastercontainer:
    name: nextcloud_aio_mastercontainer
```
`docker-compose up -d`

For Ubuntu 20.04 - use version: "3.4"
.
**CLOUDFLARE WEBSITE SETUP**
Create a Tunnel with settings:
Subdomain: cloud
Domain: battlezilla.com
Type: http
URL: 192.168.1.31:11000

![image](https://github.com/nextcloud/all-in-one/assets/52236863/140ac3d3-dae9-4c6d-b1e0-d7da38cd4e9b)

MY PORTAINER 
![image](https://github.com/nextcloud/all-in-one/assets/52236863/a262e5e1-2e05-440e-acfc-e684863e1300)

LOGIN  - **https**://192.168.1.31:8080

Let me know if it is ok for you.

**Reply by H3is3nb3rg404** [reply]
Ok here is where I stand currently. I ran the Cloudflred docker container on my Synology NAS. But after having issues and it being a bit slower, I started from scratch and installed it on my Unraid server. So, I currently have the Nextcloud AIO running and I'm able to connect to it using my purchased domain because the old Cloudflared is still on my NAS.
I'm newer to all of this and don't really understand the docker-compose.yml 100%
Do I need to create a new Cloudflared docker in Unraid?

Thank you for all this help!

**Reply by Sheron-i** [reply]
Hi, I'm trying to use your instructions for my Ugreen NAS running on stripped Debian 12. the only change i made was my cloudflare tunnel pointing to 8080 instead of 11000 as i did not get any output from 11000. I keep getting the "The login is blocked since Nextcloud is running." message, tried stopping apache container and tried using the https://yourdomain.tld/settings/admin/overview path, but it keeps rerouting me to the https://yourdomain.tld/login page with the same message. Can you help me out? 

Note: installing using portainer and let me know if you need any logs.

**Reply by vawaver** [reply]
Hi!
Thanks for trying out the guide — but I have to mention that I haven’t used this setup for quite a while. 
I ran into Cloudflare limitations, especially with larger file uploads, and eventually moved on to a different approach.

Also, for running the stack, I strongly recommend using a `docker-compose.yml` file instead of Portainer’s Stack feature. The main issue with Portainer is that after updating containers outside of Portainer (which often happens), the stack can disappear from the Portainer interface — making it difficult to manage or track. With `docker-compose.yml`, everything stays under your control and versioned.

Since your setup differs a bit (especially with Portainer and Cloudflare Tunnel on a custom port), I recommend posting your current issue on the official Nextcloud community forum:
👉 [https://help.nextcloud.com/](https://help.nextcloud.com)

There’s a helpful group of people there who will likely assist you with up-to-date solutions tailored to your configuration. 
Good luck!

**Reply by jsolivellase** [reply]
Hi!

I faced issues configuring it some time ago, but I managed to solve it. I have created a [repository](https://github.com/jsolivellase/nextcloud-aio-cloudflare-tunnel) with my compose file and instructions on how to configure everything. You can take a look if you want. I hope it helps.

Cheers!

**Reply by IMNdi** [reply]
> Hi!
> 
> I faced issues configuring it some time ago, but I managed to solve it. I have created a [repository](https://github.com/jsolivellase/nextcloud-aio-cloudflare-tunnel) with my compose file and instructions on how to configure everything. You can take a look if you want. I hope it helps.
> 
> Cheers!

It's not working and I have already chewed through my desk, my right hand and half of my cat. Nextcloud is the worst thing to host, I swear. I have like 50 containers, and this is giving me an aneurism.

Made a folder, cloned your repo, made the network, docker composed up.
Images got pulled, NC started, connected to localhost:8080, offered a login, went ahead.
Waited for containers to start, all is well. No errors, no warnings, docker is up, apache is listening.
Went on my domain.tld, gateway error 502
Went on local machine, localhost:8080 works, localhost:11000 returns 302 found and bounces me on domain.tld/login which is a 502.

The weirdness is that if I try domain.tld from inside the same machine as NC, I get a malformed, uninitialized page that says "nextcloud - a safe home for all your data" with a lot of controls with display:hidden in them. I don't even know.

I'm at my 50th attempt to make a NC instance. I only succeeded once, after cloning someone else's configuration, it was a tailscale build with its own tailscale. It' still up, but I wanted to have a cloudflare tunnel for it.

I'll take _any_ clue

**Reply by jsolivellase** [reply]
If I am remembering well, 502 error happens when tunnel is not properly configured. I suggest you check the tunnel logs with something like:
`docker logs -f nextcloud-aio-cloudflare-tunnel-cloudflare-tunnel-1`

I guess you will see some kind of error to diagnose what is happening.

**Reply by IMNdi** [reply]
`2025-10-25T15:01:35Z INF Updated to new configuration config="{\"ingress\":[{\"hostname\":\"[correctdomain.org]\", \"originRequest\":{}, \"service\":\"http://nextcloud-aio-apache:11000\"}, {\"service\":\"http_status:404\"}], \"warp-routing\":{\"enabled\":false}}" version=5`

Looks fine to me. There are no errors and it has written my domain into the config of the AIO, it looks like the AIO container is aware of the remote domain.

Nextcloud container has this in its init logs:
`System config value overwrite.cli.url set to string [https://[correctdomain.org]/⁠](https://[correctdomain.org]/)`

the only vaguely errory thing in any of the logs is

```
2025-10-25T15:01:33Z ERR Cannot determine default origin certificate path. No file cert.pem in [~/.cloudflared ~/.cloudflare-warp ~/cloudflare-warp /etc/cloudflared /usr/local/etc/cloudflared]. You need to specify the origin certificate path by specifying the origincert option in the configuration file, or set TUNNEL_ORIGIN_CERT environment variable originCertPath=

025-10-25T15:01:33Z INF Tunnel connection curve preferences: [X25519MLKEM768 CurveP256] connIndex=0 event=0 ip=[some IP]

2025/10/25 15:01:33 failed to sufficiently increase receive buffer size (was: 208 kiB, wanted: 7168 kiB, got: 416 kiB). See [https://github.com/quic-go/quic-go/wiki/UDP-Buffer-Sizes⁠](https://github.com/quic-go/quic-go/wiki/UDP-Buffer-Sizes) for details.
```

**Reply by IMNdi** [reply]
Addendum:

You are not going to believe this, but I went to eat and when I came back, it works. I ...

Does it need like 30 mins to work? 

Anyway, thanks for the prompt reply, I guess the configuration is fine, it was something cloud-side that needed to expire.

**Reply by der-delios** [reply]
Has one of you gotten Talk to work behind the cloudflare tunnel?

**Reply by Abraham2591** [reply]
> docker-compose up -d

I used playit.gg to create an UDP tunnel for that port, TCP works over cloudflare so I used cloudflare for the rest and playit.gg for UDP only, I need to install another instance though and the current docker AIO v13.1.0 is broken since it does not even create the apache container, been trying a couple days without luck, if anyone has an up to date yml apprach to install aio docker please let me know(my forte is not networking so sad)

**Comment by vawaver** [comment]
I don't use Unraid myself, but I assume it is possible to create a VM. 
In this VM, you need to first install **Ubuntu Server 22.04** as the primary operating system. 
In Ubuntu, you install Docker and Docker Compose support. For that, you can use a script from:
https://gitlab.com/bmcgonag/docker_installs -> install_docker_nproxyman.sh

Then you follow my tutorial above.

**Reply by H3is3nb3rg404** [reply]
I have it running on Ubuntu 22.04 but I cannot get it to work with the Nextcloud Android App. I keep getting the Malformed server configuration.

Does anyone have any ideas what I can do to correct this?
Thank you

**Comment by vawaver** [comment]
Does your Nexcloud AIO work on your domain (subdomain) via your web browser?
If it does, try to check your connection settings for the Nextcloud Android app.
Or create a new topic for that issue.

**Reply by H3is3nb3rg404** [reply]
Yes, the domain works via web browser so I will have to open a new topic.
Thank you for all your help!

**Comment by ccrsxx** [comment]
Here's my docker compose combining the nextcloud and cloudflared tunnel in one file. This configuration is for windows, so you might need change something on the volume if you use linux or mac.

One thing that I really struggled with was to get the cloudflared tunnel working. For some reason, I need to use the host IP address instead of network alias or the container name for the nextcloud to make it work on the cloudflared tunnel dashboard, thanks to @vawaver on this thread.

Cloudflared tunnel config with my host IP `192.168.1.3`:

![image](https://github.com/nextcloud/all-in-one/assets/55032197/8c5e4567-93ad-4189-bc1e-e9158d1bfb68)

`compose.yaml`
```yaml
services:
  app:
    image: nextcloud/all-in-one:latest
    restart: unless-stopped
    container_name: nextcloud-aio-mastercontainer
    volumes:
      - nextcloud_aio_mastercontainer:/mnt/docker-aio-config
      - //var/run/docker.sock:/var/run/docker.sock:ro
    ports:
      - 80:80
      - 8080:8080
      - 8443:8443
    environment:
      - APACHE_PORT=11000
      - APACHE_IP_BINDING=0.0.0.0
      - NEXTCLOUD_DATADIR=/run/desktop/mnt/host/d/ncdata

  tunnel:
    image: cloudflare/cloudflared:latest
    restart: unless-stopped
    command: tunnel run
    environment:
      - TUNNEL_TOKEN=your-cloudflared-tunnel-token

volumes:
  nextcloud_aio_mastercontainer:
    name: nextcloud_aio_mastercontainer
```

**Reply by GDRPrint** [reply]
Thank you for your thoughts and posting this.yaml. I had Nextcloud up and running fine, only starting having issues when configuring the email to the webhost.
During the addition of the host email links to Cloudflare I lost access to Nextcloud,
everything is running, even get notice that Nextcloud is running, but cannot login.
I'm running a linux server.

**Reply by ericmclachlan** [reply]
@ccrsxx : Thank you. I really appreciate you sharing this with us. Particularly the screenshot of Cloudflare configuration really helped put me on the right path.

**Comment by Tony2k** [comment]
I tried the same as your but I'm not able to end the configuration, this is the message:

![NC](https://github.com/nextcloud/all-in-one/assets/16784561/560dd2fc-4681-4ce0-bd2c-03f8de64dd7b)

**Reply by aaronsbytes** [reply]
If cloudflare tunnel is too much of a problem, you may consider changing your nameservers to cloudflare. This way you can just check the proxy toggle there and have all your http/https traffic proxied through cloudflare. That's what i did, and it works perfectly.

**Reply by GDRPrint** [reply]
Ok thank you, looking into that option.

**Comment by elecktro1337** [comment]
Hello guys! What about working tunnel with STUN or just NextCloud Talk?
